lib/ask/survey.ex

```diff
@@ -593,6 +595,44 @@ defmodule Ask.Survey do
     success_rate(completed_respondents, exhausted_respondents)
   end
 
+  # Calculates the daily sucess rate since the survey was launched until it
+  # ended or today if it's running.
+  def success_rate_history(%{started_at: nil}), do: []
+
+  def success_rate_history(survey) do
+    completed_dispositions = Respondent.completed_dispositions(survey.count_partial_results)
+    completed = survey |> respondents_history(completed_dispositions)
+
+    exhausted_dispositions = Respondent.metrics_final_dispositions(survey.count_partial_results)
+    exhausted = survey |> respondents_history(exhausted_dispositions)
+
+    running_date_range(survey)
+    |> Enum.map(fn date ->
+      {date, completion_rate(completed[date], exhausted[date]) * 100}
+    end)
+  end
+
+  # Counts respondents with specified dispositions with daily granularity.
+  defp respondents_history(survey, dispositions) do
+    survey
+    |> assoc(:respondents)
+    |> select({fragment("DATE(COALESCE(completed_at, updated_at))"), fragment("COUNT(*)")})
+    |> group_by(fragment("DATE(COALESCE(completed_at, updated_at))"))
```

Comment:
I started by using `completed_at` but it seems that it used to not be defined, and most of the test suite (written in 2016) don't define the column, and rely on `updated_at` instead. There is even a test that we don't crash when the disposition is completed but `completed_at` is nil!

Reply:
Yes, this sounds odd but at the same time, it sounds familiar to me 🤷 . Maybe you can put this comment in the code, just to increase these lines' readability.